providers/common/ciphers/cipher_common.c

```diff
@@ -387,6 +383,21 @@ int cipher_generic_set_ctx_params(void *vctx, const OSSL_PARAM params[])
     return 1;
 }
 
+int cipher_generic_initiv(PROV_CIPHER_CTX *ctx, const unsigned char *iv,
+                          size_t ivlen)
+{
+    if (ivlen != ctx->ivlen
+        || ivlen > sizeof(ctx->iv)) {
+        ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_IVLEN);
+        return 0;
+    }
+    ctx->iv_set = 1;
+    ctx->ivlen = ivlen;
```

Comment:
(as far as I can read, you *shouldn't* set `ctx->ivlen` here either way, because that carries information of what the IV length *should* be, according to information you have from algorithm setup, right?)

Reply:
I will remove the line - but the question is not so straightforward.. it is really set up with a default iv length - if this could be changed then it would make sense (but that is not how we are caliing it).